catroid/src/main/java/org/catrobat/catroid/utils/CrashRecovering.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.utils;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+import android.util.Log;
+
+import com.google.gson.Gson;
+
+import org.catrobat.catroid.ui.MainMenuActivity;
+
+public class CrashRecovering {
+	private static final String TAG = CrashRecovering.class.getSimpleName();
+
+	public static final String RECOVERED_FROM_CRASH = "RECOVERED_FROM_CRASH";
+	public static final String EXCEPTION_FOR_REPORT = "EXCEPTION_FOR_REPORT";
+
+	public static void checkIfCrashRecoveryAndFinishActivity(final Activity activity) {
+		if (isRecoveredFromCrash(activity)) {
+			sendUnhandledCaughtException(activity);
+			if (!(activity instanceof MainMenuActivity)) {
+				activity.finish();
```

Comment:
> also.. cant we just return to the main menu activity?

Great idea. I've tried to go back to the MainMenuActivity by using
```java
Intent intent = new Intent(getApplicationContext(), MainMenuActivity.class);
intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
intent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
startActivity(intent);
```
and all flag variations. There are still some problems and I need to do more testing.
Meanwhile, do you have this in mind?